### PR TITLE
Reduce renders from 5 to 1

### PR DESCRIPTION
**Pull request by istarkov**
## Description

1. What is this PR about (link the issue and add a short description)

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/react-sdk/src/pubsub/create.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { createNanoEvents } from "nanoevents";
 import { useCallback, useEffect, useRef } from "react";
+// eslint-disable-next-line camelcase
+import { unstable_batchedUpdates } from "react-dom";
```
Comment (kof) [inline packages/react-sdk/src/pubsub/create.ts:4]: just map it to batchedUpdates

File: packages/react-sdk/src/pubsub/create.ts [hunk 2]
```diff
@@ -10,6 +12,25 @@ export const createPubsub = <PublishMap>() => {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const emitter = createNanoEvents<Record<any, any>>();
 
+  type Task = () => void;
+  let rafHandler: ReturnType<typeof requestAnimationFrame>;
```
Comment (kof) [inline packages/react-sdk/src/pubsub/create.ts:16]: rafId

File: packages/react-sdk/src/pubsub/create.ts [hunk 3]
```diff
@@ -10,6 +12,25 @@ export const createPubsub = <PublishMap>() => {
...
+  const tasks: Task[] = [];
+
+  const processArray = () => {
+    unstable_batchedUpdates(() => {
```
Comment (kof) [inline packages/react-sdk/src/pubsub/create.ts:20]: why do you actually need this? why is just using raf no enough?

File: packages/react-sdk/src/pubsub/create.ts [hunk 4]
```diff
@@ -10,6 +12,25 @@ export const createPubsub = <PublishMap>() => {
...
+      for (const task of tasks) {
+        task();
+      }
+      tasks.length = 0;
```
Comment (kof) [inline packages/react-sdk/src/pubsub/create.ts:24]: quite a hack

File: packages/react-sdk/src/pubsub/create.ts (hunk 3 above)
Comment (kof) [reply packages/react-sdk/src/pubsub/create.ts:20]: discussed in dm, lets add a comment why we need to call batchedUpdates

**Review by kof — APPROVED** [review]

File: packages/react-sdk/src/pubsub/create.ts (hunk 3 above)
Comment (TrySound) [reply packages/react-sdk/src/pubsub/create.ts:20]: React 18 doesn't have unstable_batchedUpdates. All updates are batched.

**Review by TrySound — APPROVED** [review]

**Pull request merged**